src/collective/mustread/tracker.py

```diff
@@ -91,11 +103,117 @@ def most_read(self, days=None, limit=None):
         for record in self.query_all(query):
             yield api.content.get(UID=record.uid)
 
-    def schedule_must_read(self, obj, userids=None, deadline=None):
-        raise NotImplementedError()
+    def schedule_must_read(self, obj, userids, deadline):
+        # get existing must read items for this object
+        qry = self._read(uid=utils.getUID(obj))
+        existing_users = [m.userid for m in qry.all()]
+
+        path = '/'.join(obj.getPhysicalPath())
+        uid = utils.getUID(obj)
+        hostname = utils.hostname()
+        now = datetime.utcnow()
+        current_user = api.user.get_current().id
+        new_users = []
+        for userid in userids:
+            if userid in existing_users:
+                # skip uses that already read the item
+                # or have an open mustread request
+                log.info('user {0} already has read / has to read {1}'.format(
+                    userid, path))
+                continue
+            # @guido: should we change the deadline for existing must_reads?
```

Comment:
OK but then you'll have to also update the interface contract. Also it then does not make sense to return a user list anymore, since now this will axiomatically apply to all the userids you passed in.